File: Easy/JavaScript/118. Pascal's Triangle.js
var generate = function(numRows) {
    let pas = [];

    for(let i = 0; i < numRows; i++){
        pas[i] = new Array(i + 1);
        pas[i][0] = 1;
        pas[i][i] = 1;
        for(let j = 1; j < i; j++){
            pas[i][j] = pas[i - 1][j - 1] + pas[i - 1][j];
        }
    }

    return pas;
};

File: Easy/Java/118. Pascal's Triangle.java
class Solution {
    public List<List<Integer>> generate(int numRows) {
        List<List<Integer>> pas = new ArrayList<>();

        for(int i = 0; i < numRows; i++){
            List<Integer> row = new ArrayList<>();
            row.add(1);
            for(int j = 1; j < i; j++){
                row.add(pas.get(i - 1).get(j - 1) + pas.get(i - 1).get(j));
            }
            if(0 < i)
                row.add(1);
            pas.add(row);
        }

        return pas;
    }
}
